packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -135,38 +200,69 @@ class TransferRootsDb extends TimestampedKeysDb<TransferRoot> {
       this.logger.warn('expected transfer root object for timestamped key')
       return
     }
-    const transferRootHash = transferRoot.transferRootHash
+    const transferRootId = transferRoot.transferRootId
     const key = this.getTimestampedKey(transferRoot)
     if (!key) {
       this.logger.warn('expected timestamped key. incomplete transfer root:', JSON.stringify(transferRoot))
       return
     }
-    if (!transferRootHash) {
-      this.logger.warn(`expected transfer root hash for timestamped key. key: ${key} incomplete transfer root: `, JSON.stringify(transferRoot))
+    if (!transferRootId) {
+      this.logger.warn(`expected transfer root id for timestamped key. key: ${key} incomplete transfer root: `, JSON.stringify(transferRoot))
       return
     }
-    const item = await this.subDb.getById(key)
+    const item = await this.subDbTimestamps.getById(key)
     const exists = !!item
     if (!exists) {
-      const value = { transferRootHash }
+      const value = { transferRootId }
       return { key, value }
     }
   }
 
-  async update (transferRootHash: string, transferRoot: Partial<TransferRoot>) {
-    const logger = this.logger.create({ root: transferRootHash })
+  async getRootHashKeyValueForUpdate (transferRoot: Partial<TransferRoot>) {
+    if (!transferRoot) {
+      this.logger.warn('expected transfer root object for root hash key')
+      return
+    }
+    const transferRootId = transferRoot.transferRootId
+    const key = transferRoot.transferRootHash
+    if (!key) {
+      this.logger.warn('expected root hash key. incomplete transfer root:', JSON.stringify(transferRoot))
+      return
+    }
+    if (!transferRootId) {
+      this.logger.warn(`expected transfer root id for root hash key. key: ${key} incomplete transfer root: `, JSON.stringify(transferRoot))
+      return
+    }
+    const item = await this.subDbRootHashes.getById(key)
+    const exists = !!item
+    if (!exists) {
+      const value = { transferRootId }
+      return { key, value }
+    }
+  }
+
+  async update (transferRootId: string, transferRoot: Partial<TransferRoot>) {
+    await this.tilReady()
+    const logger = this.logger.create({ root: transferRootId })
     logger.debug('update called')
-    transferRoot.transferRootHash = transferRootHash
+    transferRoot.transferRootId = transferRootId
     const timestampedKv = await this.getTimestampedKeyValueForUpdate(transferRoot)
     const promises: Array<Promise<any>> = []
```

Comment:
probably my biggest nitpick ever :) -- Swap the order of the lines `const timestampedKv = ...` and `const promises: ...` since now the promises are used in more than just that `if (timestampedKv)` line.

Reply:
Updated in https://github.com/hop-protocol/hop/pull/318/commits/48bcdaf9078daa59da9e74ca6af291aec5d1489c